### ISSUE TITLE
In-page navigation (sticky left)

### ISSUE THREAD

**Issue by mattrosno** [carbon-design-system/carbon-platform#385]
Reference pattern: https://www.ibm.com/standards/carbon/patterns/in-page-navigation

Use Dotcom's TOC: https://www.ibm.com/standards/carbon/react/?path=/story/components-table-of-contents--default

**Comment by alisonjoseph** [comment]
Not sure we can get ` @carbon/ibmdotcom-react` to play nice <img width="818" alt="Screen Shot 2022-03-18 at 1 47 52 PM" src="https://user-images.githubusercontent.com/2753488/159065463-c74303ef-4664-4730-9059-35d2e0813ed1.png">

**Comment by alisonjoseph** [comment]
After looking into using the dot com component we realized it wouldn't work for us due to v10/v11 compatibility issues, a long with the amount of layout overrides we would have to do because it is so opinionated on layout. 

@kingtraceyj is there a detailed spec for this, I didn't see it in the sketch file, but I could be missing it. Just want to be sure I have the color and spacing tokens correct, also happy to pair up quickly to tweak what I have built.
